server/app/core/dependencies/database.py

```diff
@@ -0,0 +1,31 @@
+"""Database dependencies"""
+
+from aiomysql import Connection, Pool
+from aiomysql.cursors import Cursor
+from fastapi import Depends, Request
+
+
+def _get_db_pool(request: Request) -> Pool:
+    return request.app.state.pool
+
+
+async def _get_db_connection(pool: Pool = Depends(_get_db_pool)) -> Connection:
+    conn: Connection = await pool.acquire()
+    try:
+        yield conn
+    finally:
+        # BUG: Idk if we are supposed to close the connection after every request
+        conn.close()
+
+
+async def get_connection(conn: Connection = Depends(_get_db_connection)) -> Cursor:
```

Comment:
Do we have a need for cursors?

Reply:
That's the only way afasik to communicate to db when using connection pools according to [the docs](https://aiomysql.readthedocs.io/en/latest/pool.html#pool)